lib/ansible/modules/files/file.py

```diff
@@ -566,6 +568,18 @@ def execute_touch(path, follow, timestamps):
 
         result['changed'] = changed
         result['diff'] = diff
+    else:
+        # check_mode is True, only if the file does not exist or when one or both access_time are modification_time
+        # parameters are not set to "preserve"
+        if prev_state == 'absent':
+            result['changed'] = True
+        elif module.params['access_time'] != 'preserve' or module.params['modification_time'] != 'preserve':
```

Comment:
this is handled by update_timestamp_for_file. You can simplify this and just do an if/else
```suggestion
        else:
```